addons/account/tests/test_invoice_taxes.py

```diff
@@ -427,3 +427,43 @@ def test_misc_journal_entry_tax_tags_purchase(self):
             {'balance': -100.0,     'tax_ids': [],              'tag_ids': self.tax_tag_pos.ids},
             {'balance': 1100.0,     'tax_ids': [],              'tag_ids': []},
         ])
+
+    def test_tax_calculation_foreign_currency_large_quantity(self):
+        ''' Test:
+        Foreign currency with rate of 1.1726 and tax of 21%
+        price_unit | Quantity  | Taxes
+        ------------------
+        2.82       | 20000     | 21% not incl
+        '''
+        self.env['res.currency.rate'].search([]).unlink()
```

Comment:
self.cr.execute("UPDATE res_company SET currency_id = %s WHERE id = %s", [self.currency_euro_id, company.id])

Reply:
Updated that line for the test and added the `round=False` parameter, this works better (it converts the tax base amount correctly too)